src/mappers/subject_mapper.c

```diff
@@ -62,11 +62,15 @@ static char * subject_mapper_find_user(X509 *x509, void *context, int *match) {
 		DBG("X509_get_subject_name failed");
 		return NULL;
 	}
-	return mapfile_find(filename,entries[0],ignorecase,match);
+	char* val = mapfile_find(filename,entries[0],ignorecase,match);
+	if(entries[0]) {
```

Comment:
Shouldn't we have a universal function to free `cert_info()` entries? Something like:

```
while (entries[i]) {
    free(entries[i]);
}
```

Freeing just the 0-th element looks strange… What if there would be more than one element?

Reply:
There might be a bigger issue with this one. The pattern that is used in `cert_info.c` doesn't track the possible size of the `entries` allocated and there might be other memory leaks going on with other functions in the file. I just tracked memory leaks for `subject_mapping` and ended up in the `cert_info_subject` function and took note that it only allocated a single entry. I think a re-factor will need to be done to return the actual count of entries allocated. I thought it was strange that `cert_info_subject` is using